wnfs/src/private/hamt/node.rs

```diff
@@ -376,76 +377,81 @@ where
         }
     }
 
-    #[async_recursion(?Send)]
-    async fn remove_value<'a, 'b, B: BlockStore>(
-        self: &'a Rc<Self>,
-        hashnibbles: &'b mut HashNibbles,
-        store: &B,
-    ) -> Result<(Rc<Self>, Option<Pair<K, V>>)>
+    fn remove_value<'a, B: BlockStore>(
+        self: Rc<Self>,
+        hashnibbles: &'a mut HashNibbles,
+        store: &'a B,
+    ) -> LocalBoxFuture<'a, Result<RemovedResult<K, V, H>>>
```

Comment:
tbh, I'm almost more preferring this over the macro, which hides a bit too much.

Reply:
Hm. Alright. Didn't expect that :)

I'll remove the `RemovedResult` type alias and just `#[allow(clippy...)]` though.